### DISCUSSION TITLE
Secure STM for production - Flow Needed

### DISCUSSION THREAD

**Discussion by roizano** [micropython/micropython#14305] [STM32 / Pyboard]
Hi,
I'm trying to understand what is the right and official way to take the full code that I have in my design and compile it into one BIN file.
In addition, I want this version to have non-USB capabilities - just power and no access to the source code at all.
any guide that someone can help with? I'm using a custom STM32 board design.

**Comment by IHOXOHI** [comment]
Hi,
There is models from stmicroelectronics which could be break after downloaded your code...
